From f4459b9931216bcdc1a593fda17fb4d47045513d Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Sat, 25 May 2024 10:55:26 -0700
Subject: [PATCH] ==== selftests/damon: test DAMOS tried_regions and
 {min,max}_nr_regions ====

Subject: selftests/damon: test DAMOS tried regions

This patch series fix trivial issues in DAMON selftest code and
implement new selftests for basic functionality of DAMOS tried regions
and {min,max}_nr_regions feature.  The test for max_nr_regions also test
the recovery from online tuning-caused limit violation, which was fixed
by a previous patch [1] titled "mm/damon/core: merge regions
aggressively when max_nr_regions is unmet".

The first patch fixes a trivial problem in the articial memory access
pattern generator.  Following 3 patches (2-4) implement schemes tried
regions test.  Then a couple of patches (5-6) implementing static setup
based {min,max}_nr_regions functionality test follows.  Final two
patches (7-8) implement dynamic max_nr_regions update test.

[1] https://lore.kernel.org/20240624210650.53960C2BBFC@smtp.kernel.org

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 damon_meta_changes/hiIKbBkZ | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 damon_meta_changes/hiIKbBkZ

diff --git a/damon_meta_changes/hiIKbBkZ b/damon_meta_changes/hiIKbBkZ
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.2

